src/egit.h

```diff
@@ -36,6 +36,18 @@
 #define EGIT_ASSERT_STRING_OR_NIL(val) \
     do { if (EGIT_EXTRACT_BOOLEAN(val)) EGIT_ASSERT_STRING(val); } while (0)
 
+/**
+ * Assert that VAL is an integer, signal an error and return otherwise.
+ */
+#define EGIT_ASSERT_INTEGER(val)                                        \
+    do { if (!em_assert(env, em_integerp, (val))) return em_nil; } while (0)
```

Comment:
You can also use this in https://github.com/magit/libegit2/blob/master/src/egit-status.c#L30, there's a TODO for that

Reply:
Will do! :+1: